modules/contracts/adapters/dex/examples/deploy.rs

```diff
@@ -12,8 +12,12 @@ use semver::Version;
 const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
 
 fn deploy_dex(network: ChainInfo) -> anyhow::Result<()> {
```

Comment:
@Kayanski will this deploy command automatically select the correct feature-flagged build? How are we going about that?

Reply:
No it's going to select whatever first artifacts it founds in the directory (alphabetically I guess). So that's not good. 
@Buckram123 had a solution somewhere, don't remember what he proposed

Reply:
Oh, found it. We wanted to have an interface that is not automatically generated and that would take an extra field an instantiation for specifying the chain it was supposed to go on. 
My solution was using environment variables, but that's not Great

Reply:
> Oh, found it. We wanted to have an interface that is not automatically generated and that would take an extra field an instantiation for specifying the chain it was supposed to go on. My solution was using environment variables, but that's not Great

We still use not the automatically generated the solution was:
![image](https://github.com/AbstractSDK/abstract/assets/91957742/b4dc7e35-ac68-4192-854f-5f12c7348098)

Reply:
@adairrr Thoughts on that approach? We will have to wait a bit longer due to block_info request tho. Should be still faster than manually setting it tbh